src/chainstate/stacks/index/storage.rs

```diff
@@ -109,6 +106,67 @@ pub fn fseek_end<F: Seek>(f: &mut F) -> Result<u64, Error> {
         .map_err(Error::IOError)
 }
 
+#[derive(Clone)]
+pub struct BlockHashMap {
+    next_identifier: u32,
+    map: Vec<BlockHeaderHash>
+}
+
+trait NodeHashReader {
```

Comment:
Why is this trait called a "reader" if the only method is for writing?

Reply:
It’s reading itself to a writer. Similar to if you open a read only file and write it out to a writer. So probably it’s method name should change to read_

Reply:
I changed the method name to `read_node_hash_bytes`, which I believe is the correct naming scheme for reading bytes out of a reader and into a writer (but I've never been able to be terribly consistent with naming functions). Let me know if that works, or if you think a different name is more appropriate.